src/chart/line/LineView.js

```diff
@@ -453,6 +453,12 @@ export default ChartView.extend({
         }
 
         var visualColor = getVisualGradient(data, coordSys) || data.getVisual('color');
+        if (typeof visualColor === 'function') {
+            var color = visualColor;
+            data.each(function (idx) {
+                visualColor = color(seriesModel.getDataParams(idx));
+            });
```

Comment:
We don't have to iterate all the data to get a color here.

I think there are two solutions: 

+ use the first or last data for this callback and get one color value.
+ Or pick some of the data and get a gradient color.

From my perspective, gradient color is better, but the main concern is SVG or Canvas may have an issue with too many color stops on the gradient. About this, perhaps we can pick at most 10 of the data to get the gradient.

BTW: Personally I prefer using `colorCallback` over `color` on the variable name because it's more specific.